runtime/lua/vim/_editor.lua

```diff
@@ -1000,28 +1000,59 @@ function vim.deprecate(name, alternative, version, plugin, backtrace)
   if displayed and backtrace ~= false then
     vim.notify(debug.traceback('', 2):sub(2), vim.log.levels.WARN)
   end
-  if displayed then
-    return msg
-  end
+  return displayed and msg or nil
 end
 
 --- Create builtin mappings (incl. menus).
 --- Called once on startup.
 function vim._init_default_mappings()
   -- mappings
 
-  --@private
+  ---@private
+  local function region_chunks(region)
+    local chunks = {}
+    local maxcol = vim.v.maxcol
+    for line, cols in vim.spairs(region) do
+      local endcol = cols[2] == maxcol and -1 or cols[2]
+      local chunk = vim.api.nvim_buf_get_text(0, line, cols[1], line, endcol, {})[1]
+      table.insert(chunks, chunk)
+    end
+    return chunks
+  end
+
+  ---@private
+  local function _visual_search(cmd)
+    assert(cmd == '/' or cmd == '?')
+    vim.api.nvim_feedkeys('\27', 'nx', true) -- Escape visual mode.
+    local region = vim.region(0, "'<", "'>", vim.fn.visualmode(), vim.o.selection == 'inclusive')
+    local chunks = region_chunks(region)
+    local esc_chunks = vim
+      .iter(chunks)
+      :map(function(v)
+        return vim.fn.escape(v, [[/\]])
```

Comment:
@planet36, when `cmd` is `?`, escaped characters originally were `[[?\]]`. Is there a reason to dropping this?

With current approach, `#` fails on the following example after `v$h#` with cursor on the start of the first line:

```
aa?/\bb
aa
aa?/\bb
```

Reply:
That's my mistake, thanks for mentioning it